.github/workflows/add-netlify-links.yml

```diff
@@ -1,55 +1,75 @@
-        runs-on: ubuntu-latest
-        permissions:
-            issues: write
-            contents: write
-            pull-requests: write
-            repository-projects: write
-        steps:
-          - uses: actions/checkout@v4
-          - name: Get Changed Files
-            id: changed-files
-            uses: tj-actions/changed-files@v44
-            with:
-              separator: ","
-              files: source/**
-          - name: Build Netlify Links for Changed Pages
-            id: build_page_links
-            run: |
-              new_links=""
-              base_link='https://deploy-preview-${{ github.event.number }}--mongodb-docs-csharp.netlify.app'
-              changed_files=${{ steps.changed-files.outputs.all_changed_files }}
-              files=$(echo $changed_files | tr "," "\n")
-              for file in $files; do
-                echo "processing ${file}"
-                if (! grep -s "includes/"  <<< $file) && 
-                    (! grep -s "images/"  <<< $file) && 
-                    (! grep -s "examples/"  <<< $file); then
-                  file="${file#source}"
-                  file="${file%.txt}"
-                  filenoslash="${file:1}"
-                  echo "${base_link}${file}"
-                  new_links+="<li><a href=${base_link}${file}>${filenoslash}</a></li>"
-                else
-                  echo "(file skipped)"
-                fi
-              done
-              if [ "$new_links" == "" ]; then
-                new_links="No pages to preview"
-              fi
-              echo "Final new_links string: "
-              echo "${new_links}"
-              echo "staging_links=${new_links}" >> "$GITHUB_OUTPUT"
-          - name: Update the PR Description
-            uses: MongoCaleb/pr-description-action@master
-            with:
-                regex: "<!-- start insert-links -->.*<!-- end insert-links -->"
-                appendContentOnMatchOnly: true
-                regexFlags: is
-                content: "<!-- start insert-links -->\n${{ steps.build_page_links.outputs.staging_links }}\n<!-- end insert-links -->"
-                token: ${{ secrets.GITHUB_TOKEN }}
+  get-pr-changes:
+    name: Get Changed Files & Update PR Description
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      contents: write
+      pull-requests: write
+      repository-projects: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Changed Files
+        id: changed-files
+        # pin to a specific commit to ensure stability
+        uses: tj-actions/changed-files@c65cd883420fd2eb864698a825fc4162dd94482c
+        with:
+          separator: ","
+          files: source/**
+      - name: Build Netlify Links for Changed Pages
+        id: build_page_links
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: |
+          # Function to validate file paths
+          validate_file_path() {
```

Comment:
When I checked the source at tj-actions/changed-files, I _think_ I found that this check is already there. In other words, what it outputs are only truly changed files. It makes sense, since GitHub can't report on a changed file unless it is a valid file. 
This check might be a good safety net, but also complicates the steps and may be a source of issues if something else changes.

Reply:
As long as this is an action whose code we don't monitor, it seems like there's always a risk of validation code being changed or removed. If we don't think the convenience tradeoff is worth it, I can remove it.

Reply:
Fair enough. I'm perhaps too trusting.